torch/nn/functional.py

```diff
@@ -5066,7 +5066,7 @@ def multi_head_attention_forward(
     value: Tensor,
     embed_dim_to_check: int,
     num_heads: int,
-    in_proj_weight: Tensor,
+    in_proj_weight: Optional[Tensor],
```

Comment:
I don't think the function actually handles None in_proj_weight ? In particular `_in_projection_packed` it calls does not.

Reply:
`in_proj_weight` can be `None` since it's unused when `use_separate_proj_weight=True`. In fact, the module form sets it as None for this case:

https://github.com/pytorch/pytorch/blob/e5282c3cb8bf6ad8c5161f9d0cc271edb9abed25/torch/nn/modules/activation.py#L934-L943

`_in_projection_packed` does indeed assume that it's set.

Reply:
Ho my bad! Sounds good then!